### ISSUE TITLE
KSQL returning NULL when trying to EXTRACTJSONFIELD

### ISSUE THREAD

**Issue by Oegma2** [confluentinc/ksql#2039]
We are on version 5.0.0 OSS edition

Trying to create a stream that includes partial data from a stream's record created form a topic and the data is json. Below is an example of the data that we looking at
```
{
	"geoip": {
		"city_name": "some_city",
                "country": "some_country",
                "country_code": "12345"
        }
} 
```

We created a stream from the topic and used varchar for geoip and now want to extract the extra.unitName from the steam using EXTRACTJSONFIELD - below is an example

```
select EXTRACTJSONFIELD(geoip, '$.city_name') as city_name from unitDataStream limit 5;
```

But we getting null returns for this - and if you just select geoip, the data is there - so not that the geoip field is empty, it contains subfields and the city_name is definitely populated as well

I then pull the latest KSQL from git on branch 5.0.x and test it after compiling it with mvn install - still the same problem

Then pull latest KSQL from branch 5.1.x and tested it - also still the same issue

Any other way I can help test this function to ensure it working? Any extra details I need to provide, will be happy to do so.

We want to eventually use this function to extra sub-fields within json for complex KSQL window queries and we using the work-around at the moment with a map<string,string> but it becomes even more complex if you json sub-fields contains lower levels...that map<string,....> get super complex. Will be best to use a build in function like EXTRACTJSONFIELD to extra those sub fields - reads easier as well

**Comment by apurvam** [comment]
Can you share the KSQL server logs?

**Comment by beckxie** [comment]
Hi @Oegma2 , 
I create a new issue. #2095 
Is it the same?

**Comment by rodesai** [comment]
@Oegma2 this seems to be fixed in KSQL 5.3. You can either:
   - upgrade to this version
   - declare a STRUCT schema for geoip instead of VARCHAR, and access the child field as a struct member.
